api/v1alpha1/physical_machine_chaos_webhook.go

```diff
@@ -77,5 +81,317 @@ func (in *PhysicalMachineChaosSpec) Validate(root interface{}, path *field.Path)
 		}
 	}
 
+	if skipConfigCheck {
+		return allErrs
+	}
+
+	var validateConfigErr error
+	switch in.Action {
+	case PMStressCPUAction:
+		validateConfigErr = validateStressCPUAction(in.StressCPU)
+	case PMStressMemAction:
+		validateConfigErr = validateStressMemAction(in.StressMemory)
+	case PMDiskWritePayloadAction:
+		validateConfigErr = validateDiskPayloadAction(in.DiskWritePayload)
+	case PMDiskReadPayloadAction:
+		validateConfigErr = validateDiskPayloadAction(in.DiskReadPayload)
+	case PMDiskFillActionAction:
+		validateConfigErr = validateDiskFillAction(in.DiskFill)
+	case PMNetworkCorruptAction:
+		validateConfigErr = validateNetworkCorruptAction(in.NetworkCorrupt)
+	case PMNetworkDuplicateAction:
+		validateConfigErr = validateNetworkDuplicateAction(in.NetworkDuplicate)
+	case PMNetworkLossAction:
+		validateConfigErr = validateNetworkLossAction(in.NetworkLoss)
+	case PMNetworkDelayAction:
+		validateConfigErr = validateNetworkDelayAction(in.NetworkDelay)
+	case PMNetworkPartitionAction:
+		validateConfigErr = validateNetworkPartitionAction(in.NetworkPartition)
+	case PMNetworkDNSAction:
+		validateConfigErr = validateNetworkDNSAction(in.NetworkDNS)
+	case PMProcessAction:
+		validateConfigErr = validateProcessAction(in.Process)
+	case PMJVMExceptionAction:
+		validateConfigErr = validateJVMExceptionAction(in.JVMException)
+	case PMJVMGCAction:
+		validateConfigErr = validateJVMGCAction(in.JVMGC)
+	case PMJVMLatencyAction:
+		validateConfigErr = validateJVMLatencyAction(in.JVMLatency)
+	case PMJVMReturnAction:
+		validateConfigErr = validateJVMReturnAction(in.JVMReturn)
+	case PMJVMStressAction:
+		validateConfigErr = validateJVMStressAction(in.JVMStress)
+	case PMJVMRuleDataAction:
+		validateConfigErr = validateJVMRuleDataAction(in.JVMRuleData)
+	case PMClockAction:
+		validateConfigErr = validateClockAction(in.Clock)
+	}
+
+	if validateConfigErr != nil {
+		allErrs = append(allErrs,
+			field.Invalid(path.Child("spec"), in,
+				validateConfigErr.Error()))
+	}
+
 	return allErrs
 }
+
+func validateStressCPUAction(spec *StressCPUSpec) error {
+	if spec.Load == 0 {
+		return errors.New("load can't be 0")
+	}
+
+	if spec.Workers == 0 {
+		return errors.New("workers can't be 0")
+	}
+
+	return nil
+}
+
+func validateStressMemAction(spec *StressMemorySpec) error {
+	if len(spec.Size) == 0 {
+		return errors.New("size is required")
+	}
+
+	return nil
+}
+
+func validateDiskPayloadAction(spec *DiskPayloadSpec) error {
+	if spec.PayloadProcessNum == 0 {
+		return errors.New("payload-process-num can't be 0")
+	}
+
+	if len(spec.Size) == 0 {
+		return errors.New("size is required")
+	}
```

Comment:
Do we need to check the unit of `size` field(e.g. MB, MiB)?

Reply:
done